dimod/constrained.py

```diff
@@ -618,6 +619,88 @@ def from_file(cls, fp: Union[BinaryIO, ByteString]) -> "ConstrainedQuadraticMode
 
         return cqm
 
+    def iter_violations(self, sample_like, *, skip_satisfied: bool = False, clip: bool = True,
+                        ) -> Iterator[Tuple[Hashable, Bias]]:
+        """Yield constraint labels and the amount the constraints are violated.
+
+        Args:
+            sample_like: A sample over the CQM variables.
+            skip_satisfied: If True, samples that are not violated are skipped.
+            clip: If False, then negative violations will be returned.
+
+        Yields:
+            A 2-tuple containing the constraint label and the amount of
+            constraints violation.
+
+        Example:
+
+            Construct a constrained quadratic model.
+
+            >>> i, j, k = dimod.Binaries(['i', 'j', 'k'])
+            >>> cqm = dimod.ConstrainedQuadraticModel()
+            >>> cqm.add_constraint(i + j + k == 10, label='equal')
+            'equal'
+            >>> cqm.add_constraint(i + j <= 15, label='less equal')
```

Comment:
"label='less equal'" can we give a nicer label? Perhaps "less or equal to 15"? Makes the printed yield clearer below